### PR TITLE
[New RM] - Add Tavily search

### PR DESCRIPTION
**Pull request by zenith110**
This PR aims to accomplish the following:
- Add Tavily search
- Update examples to include Tavily retriever
- Update readme.md to include TavilyRM.

Input + output is attached.
[tavily-run-input-output.txt](https://github.com/user-attachments/files/16731057/tavily-run-input-output.txt)

Thank you.

### CODE REVIEW

File: knowledge_storm/rm.py [hunk 1]
```diff
@@ -691,3 +691,124 @@ def forward(
 
         return collected_results
 
+class TavilySearchRM(dspy.Retrieve):
+    """Retrieve information from custom queries using Tavily."""
```
Comment (Yucheng-Jiang) [inline knowledge_storm/rm.py:695]: would you please add a link to the search engine API documentation?

**Review by Yucheng-Jiang — COMMENTED** [review]
Thanks for working on this! Could you provide storm_polished_article.txt associated with your input-output example? Would love to see the actual generate article to assess the quality.

**Comment by zenith110** [comment]
Hi @Yucheng-Jiang, attached are two polished articles. One is for pokemon(the ascii e displays as a unknown character which is strange), and the polished for the original one based off the input-output of shiba inu.
[storm_gen_article_polished_shiba.txt](https://github.com/user-attachments/files/16736781/storm_gen_article_polished_shiba.txt)
[storm_gen_article_polished_pokemon.txt](https://github.com/user-attachments/files/16736783/storm_gen_article_polished_pokemon.txt)

**Review by Yucheng-Jiang — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**